### PR TITLE
Allow displaying multiple apps queried by name or selector

### PR DESCRIPTION
**Pull request by muenchdo**
I would like to propose this change to the way Argo CD applications are queried and displayed for an entity. Instead of assuming a 1:1 mapping between an entity and Argo CD application, this allows defining a label selector that can be used to retrieve a list of matching applications.

It ends up looking like this:
Screen Shot 2020-12-28 at 16 35 21

Making it a draft PR for now, as the styling definitely needs some love and I'm not too happy with that weird [type switching](https://github.com/RoadieHQ/backstage-plugin-argo-cd/compare/main...muenchdo:main#diff-dc9603413065e1b754b41f1316a7eed61f0009b2107ddf34379548a5800229b9R137-R148).

### CODE REVIEW

**Comment by muenchdo** [comment]
Would love to hear your opinion on this, @mcalus3!

**Comment by mcalus3** [comment]
I think it looks great :) You can think about making it an expandable panel if it would exceed some certain height in the widget (to not clutter the overview page with so much information until user explicitely expands it. [Github Insights plugin](https://github.com/RoadieHQ/backstage-plugin-github-insights) has it implemented) but it's not necessary.

**Comment by muenchdo** [comment]
For the sake of simplicity (and to keep the changeset relatively small), I would not make the panel expandable right now. I'm happy to contribute more improvements in the future though 🙂 I'll mark this as ready for review then.

File: src/components/ArgoCDDetailsWidget.tsx [hunk 1]
```diff
@@ -159,10 +109,10 @@ export const ArgoCDDetailsWidget = ({ entity }: { entity: Entity }) => {
 };
 
 const ArgoCDDetails = ({ entity }: { entity: Entity }) => {
-  const appName = useArgoCDAppData({ entity });
+  const { appName, appSelector } = useArgoCDAppData({ entity });
 
-  const { loading, value, error } = useAppDetails({
-    appName,
+  const { loading, value, error } = !!appName ? useAppDetails({ appName }) : useListAppDetails({
```
Comment (mcalus3) [inline src/components/ArgoCDDetailsWidget.tsx:114]: It's illegal to invoke hooks conditionally. I suggest moving this switch into the hook useAppDetails

**Review by mcalus3 — COMMENTED** [review]
Could you also add to the readme some examples how to use new annotations?

File: src/components/ArgoCDDetailsWidget.tsx (hunk 1 above)
Comment (muenchdo) [reply src/components/ArgoCDDetailsWidget.tsx]: Resolved in c0f8821f1ca554966473c7be6e04a9b4c6df9f7f.

**Review by mcalus3 — APPROVED** [review]

**Comment by mcalus3** [comment]
Thanks for your contribution 👍

**Pull request merged**